guides/release/tutorial/9.md

```diff
@@ -0,0 +1,152 @@
+      </p>
+      <a href="tel:503.555.1212">+1 (503) 555-1212</a><br>
+      <a href="mailto:superrentalsrep@emberjs.com">superrentalsrep@emberjs.com</a>
+    </address>
+    <LinkTo @route="about" class="button">About</LinkTo>
+  </Jumbo>
+</div>
+```
+
+After saving, everything should look exactly the same as before, and all the tests should still pass. Very nice!
+
+<!-- TODO: screenshot of the page? -->
+
+While it may not save you a lot of characters in this case, it made the template slightly easier to read by *[encapsulating](TODO: link to encapsulating)* the implementation of the "jumbo" header into its own component, which allowed the reader to focus on things that are unique to just that page. Now, if we need to make a change to the header, we will just have to make it in a single place. Feel free to give that a try!
+
+Before we move on to the next component, let's write an automated test for our `<Jumbo>` component. Run this command in your terminal:
+
+```shell
+ember generate component-test jumbo
+```
+
+Here, we used the generator to generate a *[component test](TODO: link to component test)*. These are used to render and test a single component at a time. This is in contrast to the acceptance tests that we wrote earlier, which have to navigate and render entire pages worth of content.
+
+Let's replace the boilerplate code that was generated for us with our own test:
+
+```js {data-filename="tests/integration/components/jumbo.js" data-diff="+9,+10,+11,+12,+13,+14,+15,-16,-17,-18,-19,-20,-21,-22,-23,-24,-25,-26,-27,-28,-29,-30,-31,-32"}
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+
+module('Integration | Component | jumbo', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders the content inside a jumbo header with a tomster', async function(assert) {
+    await render(hbs`<Jumbo>Hello World</Jumbo>`);
+
+    assert.dom('.jumbo').exists();
+    assert.dom('.jumbo').containsText('Hello World');
+    assert.dom('.jumbo .tomster').exists();
+  });
+
+  test('it renders', async function(assert) {
+    // Set any properties with this.set('myProperty', 'value');
+    // Handle any actions with this.set('myAction', function(val) { ... });
+
+    await render(hbs`<Jumbo />`);
+
+    assert.equal(this.element.textContent.trim(), '');
+
+    // Template block usage:
+    await render(hbs`
+      <Jumbo>
+        template block text
+      </Jumbo>
+    `);
+
+    assert.equal(this.element.textContent.trim(), 'template block text');
+  });
+});
+```
+
+Instead of navigating to a URL, we start the test by rendering our `<Jumbo>` component on the test page. This is useful because it may otherwise require a lot of setup and interaction just to get to a page where your component is used. Component tests allows us to skip all of that and focus exclusively on testing the component by itself.
```

Comment:
```suggestion
Instead of navigating to a URL, we start the test by rendering our `<Jumbo>` component on the test page. This is useful because it may otherwise require a lot of setup and interaction just to get to a page where your component is used. Component tests allows us to skip all of that and focus exclusively on testing the component in isolation.
```